src/System.Text.Primitives/System/Text/Encoders/Ascii/Ascii_casing.cs

```diff
@@ -27,50 +27,71 @@ static Ascii()
 
             public static OperationStatus ToLowerInPlace(Span<byte> ascii, out int bytesChanged)
             {
-                for (bytesChanged = 0; bytesChanged < ascii.Length; bytesChanged++)
+                int tempBytesChanged = 0;
```

Comment:
It's interesting how much of a difference this makes (30-50% improvement).

https://www.diffchecker.com/kv75bgea
[sharplab-assembly](https://sharplab.io/#v2:C4LghgzgtgPgAgBgARwIwG4CwAoRLUB0AQgK4BmZApgE4RbY5wDMKATEgMJIDeOS/SAEYBPYJQDaAXSQQA+sAD2AGQUB3GkgC8SAHaVVQ0RNSsAHJPoCkfAcyQB5AA40wwAJYKdAZWCuSEJAAVZTUaADl9AAoAJUowABN7HQAbYS9HMB0AHhExAD4kNx1HEmAAGiR0zJyjAoVSkvKkeuBCnVbHagUAY0oICEp4oiMIAEobfl5sKysi1rEoRwAFLt7+weGxAO0ESxn+OaQoIq02xoIlSh0Ac2AACyQs5obSi6vbh4B+M9fLm/ukCBnsBzn8Pnt9mQFNQkJEFstVn0BkMRqddkh4SsekiNqinscdOgMZRFli1sjNn0ANRU8bTfY8CYM/i5Si6SgAD1a2iKjXEmMR6xRWwsTOZbjIsL0XKQBRMAHY6cyrFNlQzOtihZTtsTSYKKSMIWrbPKHM5qK4PN5fMB/AQAJI6ABuYGSbniABFXGAjcyAL5ihktPkCzUGkWnOSKFTqajiaXAUX0/YB5MzDXk3FbU6hzPCvq+k1mlzuTw+PwQAgezyUI2pqxMuxOEtW8u2gLBGM0ezUNzXIqumJxRIpNIZbKsgq80oVKoT2rAxoVFptDr6rN9JUCVUMw4E07T4Bvf4PJ7B37vAHfQ/Hj6AxcXk+F/hQmGRDM4/M69EfrV4o5FESv7htStKBpM4EzKy7IyjyxSlOIwEbhASbGoUkqRAmspIAqowoKazYWqW1oVg6zquu6Xq+M+MznsAiHrl+0jaFGISxvGnKJjR9YMnABHmpaZY2na1Z6HWOB+kAA)

Is there anything that we can do about this?
cc @AndyAyersMS